homeassistant/components/zwave_js/switch.py

```diff
@@ -62,3 +71,59 @@ async def async_turn_off(self, **kwargs: Any) -> None:
         target_value = self.get_zwave_value("targetValue")
         if target_value is not None:
             await self.info.node.async_set_value(target_value, False)
+
+
+class ZWaveBarrierEventSignalingSwitch(ZWaveBaseEntity, SwitchEntity):
+    """This switch is used to turn on or off a barrier device's event signaling subsystem."""
+
+    def __init__(
+        self,
+        config_entry: ConfigEntry,
+        client: ZwaveClient,
+        info: ZwaveDiscoveryInfo,
+    ) -> None:
+        """Initialize a ZWaveBarrierEventSignalingSwitch entity."""
+        super().__init__(config_entry, client, info)
+        self._name = self.generate_name(include_value_name=True)
+        self._state: Optional[int] = None
+
+        self._update_state()
+
+    @callback
+    def on_value_update(self) -> None:
+        """Call when a watched value is added or updated."""
+        self._update_state()
+
+    @property
+    def name(self) -> str:
+        """Return default name from device name and value name combination."""
+        return self._name
+
+    @property
+    def is_on(self) -> Optional[bool]:  # type: ignore
+        """Return a boolean for the state of the switch."""
+        return bool(self._state)
```

Comment:
This won't return None now, so the return value isn't optional. If we want it optional we should not use the `bool` copy.

Reply:
If None is returned, does that result in an "unknown" state for the entity? Maybe that is preferable.

Reply:
Not currently, but it may change in the future. I think it's more correct to return None since we don't know the state.